src/components/Input/Toggle/Toggle.styles.js

```diff
@@ -0,0 +1,125 @@
+/* stylelint-disable no-duplicate-selectors */
+import styled from "styled-components";
+import { colors, constants } from "../../..//theme";
+
+const SIZES = {
+  small: {
+    track: `
+        width: 30px;
+        height: 16px;
+        border: 1px solid;
+    `,
+    thumb: `
+        width: 10px;
+        height: 10px;
+    `,
+    activeAreaWidth: "44px",
+    offsetRange: ["3px", "17px"]
+  },
+  regular: {
+    track: `
+      width: 48px;
+      height: 24px;
+      border: 1px solid;
+    `,
+    thumb: `
+      width: 18px;
+      height: 18px;
+    `,
+    activeAreaWidth: "73px",
+    offsetRange: ["3px", "27px"]
+  }
+};
+
+const COLORS = {
+  active: {
+    border: colors.emerald.base,
+    background: colors.emerald.base,
+    thumb: colors.white.base
+  },
+  inactive: {
+    border: colors.slate,
+    background: colors.white.base,
+    thumb: colors.slate
+  },
+  activeDisabled: {
+    border: colors.diatomite,
+    background: colors.emerald.restrained,
+    thumb: colors.moonstone
+  },
+  inactiveDisabled: {
+    border: colors.diatomite,
+    background: colors.white.muted,
+    thumb: colors.moonstone
+  }
+};
+
+export const ReactToggleTrack = styled.div`
+  ${({ size }) => SIZES[size].track} padding: 0;
+  border-radius: 21px;
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  transition: background-color 0.3s ${constants.easing.easeInOutQuad};
+  background-color: ${({ variant }) => COLORS[variant].background};
+  border-color: ${({ variant }) => COLORS[variant].border};
+
+  &:focus {
+    border-color: ${colors.azure.base};
+    box-shadow: 0 0 5px 0 ${colors.azure.base};
+    outline: none;
+  }
+`;
+
+export const ActiveArea = styled.div`
+  width: ${({ size }) => SIZES[size].activeAreaWidth};
+  height: 44px;
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  cursor: ${({ disabled }) => (disabled ? "not-allowed" : "pointer")};
+
+  &:hover {
+    ${ReactToggleTrack} {
+      ${({ disabled }) =>
+        disabled ? "" : `border: 2px solid ${colors.azure.base};`};
+    }
+  }
+`;
+
+export const ReactToggle = styled.div`
+  display: flex;
+  align-items: center;
+  justify-content: flex-start;
+  position: relative;
+  background-color: transparent;
+  border: 0;
+  padding: 0;
+  user-select: none;
+`;
+
+export const ReactThumbCenteringContainer = styled.div`
+  position: absolute;
+  height: 100%;
+  top: 0;
+  left: 0;
+  display: flex;
+  flex: 1;
+  align-self: stretch;
+  align-items: center;
+  justify-content: flex-start;
+`;
+
+export const ReactToggleThumb = styled.div`
+  ${({ size }) => SIZES[size].thumb} display: flex;
```

Comment:
here let's also introduce classNames with modifiers instead of relying on props fir size, variant, value.

Reply:
The same question here: what are the benefits of using class names vs props?